### PR TITLE
Add type definition for push

### PR DESCRIPTION
**Pull request by MarkRunWu**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+   * You can register your device for receiving push notifications.
+   *
+   * @param {string} token - the device token
+   * @param {string} type - the device type (either 'ios' or 'android')
+   * @param {string} topic - the device topic, refer to application bundle
+   * identifier on iOS and application package name on Android
+   **/
+  export class PushContainer {
+    registerDevice(token: string, type: string, topic: string): Promise<string>;
+    unregisterDevice(): Promise<string>;
```
Comment (louischan-oursky) [inline index.d.ts]: The return type should be `Promise<void>`

File: index.d.ts [hunk 2]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+    sendToUser(
+      users: string[],
```
Comment (louischan-oursky) [inline index.d.ts]: `users: string | string[],`

File: index.d.ts [hunk 3]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+      notification: KVObject,
+      topic: string
```
Comment (louischan-oursky) [inline index.d.ts]: `topic?: string`

File: index.d.ts [hunk 4]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+    ): Promise<KVObject>;
```
Comment (louischan-oursky) [inline index.d.ts]: I think the return type is better to be `Promise<void>` because the actual return type is not very useful.

File: index.d.ts [hunk 5]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+
+    sendToDevice(
+      devices: string[],
```
Comment (louischan-oursky) [inline index.d.ts]: `devices: string | string[],`

File: index.d.ts [hunk 6]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+      notification: KVObject,
+      topic: string
```
Comment (louischan-oursky) [inline index.d.ts]: `topic?: string`

File: index.d.ts [hunk 7]
```diff
@@ -313,6 +314,30 @@ declare module "skygear" {
...
+    ): Promise<KVObject>;
```
Comment (louischan-oursky) [inline index.d.ts]: Return type is better to be `Promise<void>`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
@MarkRunWu

**Comment by MarkRunWu** [comment]
@louischan-oursky Updated.

**Review by louischan-oursky — APPROVED** [review]

**Comment by louischan-oursky** [comment]
Thanks!

**Comment by louischan-oursky** [comment]
@MarkRunWu I updated the installation of this package.

https://github.com/oursky/ts-skygear

Can you try it?

**Comment by MarkRunWu** [comment]
Ok, thanks

**Pull request merged**